consensus/src/experimental/ordering_state_computer.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{error::StateSyncError, state_replication::StateComputer};
+use anyhow::Result;
+use channel::Sender;
+use consensus_types::{block::Block, executed_block::ExecutedBlock};
+use diem_crypto::{hash::ACCUMULATOR_PLACEHOLDER_HASH, HashValue};
+use diem_types::ledger_info::LedgerInfoWithSignatures;
+use executor_types::{Error as ExecutionError, StateComputeResult};
+use fail::fail_point;
+use futures::SinkExt;
+use std::{boxed::Box, sync::Arc};
+
+/// Ordering-only execution proxy
+/// implements StateComputer traits.
+/// Used only when node_config.validator.consensus.decoupled = true.
+pub struct OrderingStateComputer {
+    // the channel to pour vectors of blocks into the real execution phase (will be handled in ExecutionPhase).
+    executor_channel: Sender<(Vec<Block>, LedgerInfoWithSignatures)>,
+}
+
+impl OrderingStateComputer {
+    pub fn new(executor_channel: Sender<(Vec<Block>, LedgerInfoWithSignatures)>) -> Self {
+        Self { executor_channel }
+    }
+}
+
+#[async_trait::async_trait]
+impl StateComputer for OrderingStateComputer {
+    fn compute(
+        &self,
+        // The block to be executed.
+        _block: &Block,
+        // The parent block id.
+        _parent_block_id: HashValue,
+    ) -> Result<StateComputeResult, ExecutionError> {
+        // Return dummy block and bypass the execution phase.
+        // This will break the e2e smoke test.
+        Ok(StateComputeResult::new(
+            *ACCUMULATOR_PLACEHOLDER_HASH,
+            vec![],
+            0,
+            vec![],
+            0,
+            None,
+            vec![],
+            vec![],
+            vec![],
+        ))
+    }
+
+    /// Send a successful commit. A future is fulfilled when the state is finalized.
```

Comment:
change the comment? this is more like sending ordered blocks to next phase

Reply:
Resolved.